models/demos/falcon7b/tests/test_falcon_device_perf.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from models.demos.falcon7b.tests.run_falcon_end_to_end import run_test_FalconCausalLM_end_to_end
+from models.demos.falcon7b.tt.model_config import get_model_config
+from models.perf.device_perf_utils import check_device_perf, prep_device_perf_report, run_device_perf
+from models.utility_functions import disable_compilation_reports, disable_persistent_kernel_cache
+
+
+@pytest.mark.parametrize(
+    "model_version",
+    ("tiiuae/falcon-7b-instruct",),
+    ids=["falcon_7b"],
+)
+@pytest.mark.parametrize(
+    "llm_mode, num_layers, batch, seq_len, kv_cache_len, model_config_str, expected_output_pcc, expected_k_cache_pcc, expected_v_cache_pcc, expected_inference_time",
+    (
+        ("prefill", 32, 1, 128, 0, "BFLOAT16-DRAM", 0.97, 0.99, 0.97, 0.1),
+        ("prefill", 32, 1, 1024, 0, "BFLOAT16-DRAM", 0.99, 0.99, 0.98, 0.5),
+        ("prefill", 32, 1, 2048, 0, "BFLOAT16-DRAM", 0.99, 0.99, 0.98, 1.1),
+    ),
+    ids=[
+        "prefill_seq128_bfloat16-dram",
+        "prefill_seq1024_bfloat16-dram",
+        "prefill_seq2048_bfloat16-dram",
+    ],
+)
+def test_device_perf_wh_bare_metal(
+    model_version,
+    llm_mode,
+    batch,
+    seq_len,
+    kv_cache_len,
+    expected_inference_time,
+    num_layers,
+    expected_output_pcc,
+    expected_k_cache_pcc,
+    expected_v_cache_pcc,
+    model_config_str,
+    model_location_generator,
+    get_tt_cache_path,
+    device,
+):
+    model_config = get_model_config(model_config_str, seq_len)
+    tt_cache_path = get_tt_cache_path(
+        model_version, model_subdir="Falcon", default_dir=model_config["DEFAULT_CACHE_PATH"]
+    )
+
+    disable_persistent_kernel_cache()
+    disable_compilation_reports()
+
+    run_test_FalconCausalLM_end_to_end(
+        [device],
+        model_version,
+        llm_mode,
+        batch,
+        seq_len,
+        kv_cache_len,
+        num_layers,
+        [expected_output_pcc, expected_k_cache_pcc, expected_v_cache_pcc],
+        model_config,
+        model_config_str,
+        tt_cache_path,
+        model_location_generator,
+        expected_inference_time,
+        device_perf=True,
+    )
```

Comment:
Is it possible to avoid duplicating the pcc and e2e inference targets?

Reply:
You are right. I prefer to keep PCC check for device perf test as well since the two pipelines don't run in sync so you can't reliably match device perf test with PCC you see in the latest e2e perf test. I put all PCC values in the same place.

Reply:
Device perf CI: https://github.com/tenstorrent/tt-metal/actions/runs/9596895615/job/26464771091
Model perf CI: https://github.com/tenstorrent/tt-metal/actions/runs/9595869958/job/26461748420
T3K model perf CI: https://github.com/tenstorrent/tt-metal/actions/runs/9595874577

Reply:
Looks good, thanks :)